perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3662,7 +3663,10 @@ private void setUpSynchronizationAttributesForAllSubGroups(PerunSession sess, Gr
 				log.info("ExtSource already assigned to group: {}", group);
 			}
 
-			membersQueryAttribute.setValue(baseMembersQuery.getValue().toString().replace("?", group.getShortName()));
+			//we want to set login of group
+			Attribute loginAttribute = getPerunBl().getAttributesManagerBl().getAttribute(sess, group, loginAttributeDefinition.getName());
+			if(loginAttribute.getValue() == null) throw new InternalErrorException("For purpose of setting attributes for group " + group + " we need to have not empty group login " + loginAttribute);
```

Comment:
Nit: I think you can write the message like this "...for " + group + " ...", because if you leave it like it is now, the final message will be with duplicate "group" word.

Reply:
Changed.